server/src/main/java/org/candlepin/model/AbstractHibernateObject.java

```diff
@@ -33,12 +33,17 @@
 
 /**
  * Abstract class for hibernate entities
+ *
+ * @param <T>
+ *  Entity type extending this class; should be the name of the subclass
  */
 @MappedSuperclass
 @XmlType(name = "CandlepinObject")
 @JsonIgnoreProperties(ignoreUnknown = true)
 @JsonFilter("DefaultFilter")
-public abstract class AbstractHibernateObject implements Persisted, Serializable {
+public abstract class AbstractHibernateObject<T extends AbstractHibernateObject>
```

Comment:
I made the following changes and I didn't see any warnings about `@Override` not actually doing anything.  IDEA normally warns about that too.  I don't think there are any erasure issues because the generic type in CandlepinDTO is reified by the time you get to an actual DTO implementation.

```diff
diff --git a/server/src/main/java/org/candlepin/dto/CandlepinDTO.java b/server/src/main/java/org/candlepin/dto/CandlepinDTO.java
index c89320c..fca72bb 100644
--- a/server/src/main/java/org/candlepin/dto/CandlepinDTO.java
+++ b/server/src/main/java/org/candlepin/dto/CandlepinDTO.java
@@ -90,7 +90,7 @@ public abstract class CandlepinDTO<T extends CandlepinDTO> implements Cloneable,
      * @return
      *  this CandlepinDTO instance
      */
-    public CandlepinDTO populate(CandlepinDTO<T> source) {
+    public CandlepinDTO populate(T source) {
         if (source == null) {
             throw new IllegalArgumentException("source is null");
         }
diff --git a/server/src/main/java/org/candlepin/dto/api/v1/CertificateDTO.java b/server/src/main/java/org/candlepin/dto/api/v1/CertificateDTO.java
index 752dfa2..175a04a 100644
--- a/server/src/main/java/org/candlepin/dto/api/v1/CertificateDTO.java
+++ b/server/src/main/java/org/candlepin/dto/api/v1/CertificateDTO.java
@@ -150,7 +150,7 @@ public class CertificateDTO extends TimestampedCandlepinDTO<CertificateDTO> {
      * {@inheritDoc}
      */
     @Override
-    public CertificateDTO populate(CandlepinDTO<CertificateDTO> source) {
+    public CertificateDTO populate(CertificateDTO source) {
         super.populate(source);
 
         // We're safeguarded here by both the generic definition above, and the runtime-level check
diff --git a/server/src/main/java/org/candlepin/dto/api/v1/CertificateSerialDTO.java b/server/src/main/java/org/candlepin/dto/api/v1/CertificateSerialDTO.java
index 2cd1537..9e8746a 100644
--- a/server/src/main/java/org/candlepin/dto/api/v1/CertificateSerialDTO.java
+++ b/server/src/main/java/org/candlepin/dto/api/v1/CertificateSerialDTO.java
@@ -166,7 +166,7 @@ public class CertificateSerialDTO extends TimestampedCandlepinDTO<CertificateSer
      * {@inheritDoc}
      */
     @Override
-    public CertificateSerialDTO populate(CandlepinDTO<CertificateSerialDTO> source) {
+    public CertificateSerialDTO populate(CertificateSerialDTO source) {
         super.populate(source);
 
         // We're safeguarded here by both the generic definition above, and the runtime-level check
diff --git a/server/src/main/java/org/candlepin/dto/api/v1/ConsumerTypeDTO.java b/server/src/main/java/org/candlepin/dto/api/v1/ConsumerTypeDTO.java
index a2003d4..624e58f 100644
--- a/server/src/main/java/org/candlepin/dto/api/v1/ConsumerTypeDTO.java
+++ b/server/src/main/java/org/candlepin/dto/api/v1/ConsumerTypeDTO.java
@@ -134,7 +134,7 @@ public class ConsumerTypeDTO extends TimestampedCandlepinDTO<ConsumerTypeDTO> {
      * {@inheritDoc}
      */
     @Override
-    public ConsumerTypeDTO populate(CandlepinDTO<ConsumerTypeDTO> source) {
+    public ConsumerTypeDTO populate(ConsumerTypeDTO source) {
         super.populate(source);
 
         // We're safeguarded here by both the generic definition above, and the runtime-level check
diff --git a/server/src/main/java/org/candlepin/dto/api/v1/OwnerDTO.java b/server/src/main/java/org/candlepin/dto/api/v1/OwnerDTO.java
index 7339e5e..63adbd3 100644
--- a/server/src/main/java/org/candlepin/dto/api/v1/OwnerDTO.java
+++ b/server/src/main/java/org/candlepin/dto/api/v1/OwnerDTO.java
@@ -249,7 +249,7 @@ public class OwnerDTO extends TimestampedCandlepinDTO<OwnerDTO> implements Linka
      * {@inheritDoc}
      */
     @Override
-    public OwnerDTO populate(CandlepinDTO<OwnerDTO> source) {
+    public OwnerDTO populate(OwnerDTO source) {
         super.populate(source);
 
         // We're safeguarded here by both the generic definition above, and the runtime-level check
diff --git a/server/src/main/java/org/candlepin/dto/api/v1/TimestampedCandlepinDTO.java b/server/src/main/java/org/candlepin/dto/api/v1/TimestampedCandlepinDTO.java
index 533ec7c..9135fcb 100644
--- a/server/src/main/java/org/candlepin/dto/api/v1/TimestampedCandlepinDTO.java
+++ b/server/src/main/java/org/candlepin/dto/api/v1/TimestampedCandlepinDTO.java
@@ -174,7 +174,7 @@ public abstract class TimestampedCandlepinDTO<T extends TimestampedCandlepinDTO>
      * {@inheritDoc}
      */
     @Override
-    public TimestampedCandlepinDTO populate(CandlepinDTO<T> source) {
+    public TimestampedCandlepinDTO populate(T source) {
         super.populate(source);
 
         // We're safeguarded here by both the generic definition above, and the runtime-level check
diff --git a/server/src/main/java/org/candlepin/dto/api/v1/UpstreamConsumerDTO.java b/server/src/main/java/org/candlepin/dto/api/v1/UpstreamConsumerDTO.java
index 9d87185..b2ef561 100644
--- a/server/src/main/java/org/candlepin/dto/api/v1/UpstreamConsumerDTO.java
+++ b/server/src/main/java/org/candlepin/dto/api/v1/UpstreamConsumerDTO.java
@@ -207,7 +207,7 @@ public class UpstreamConsumerDTO extends TimestampedCandlepinDTO<UpstreamConsume
      * {@inheritDoc}
      */
     @Override
-    public UpstreamConsumerDTO populate(CandlepinDTO<UpstreamConsumerDTO> source) {
+    public UpstreamConsumerDTO populate(UpstreamConsumerDTO source) {
         super.populate(source);
 
         // We're safeguarded here by both the generic definition above, and the runtime-level check
```